### PR TITLE
Misc estimator steps for scz bp

### PR DESCRIPTION
**Pull request by HLasse**
Added a bunch of estimator steps - some (probably) mainly for scz bp, but others may be more generally useful if you fancy e.g. augmentation or data-centric processing. 

Let me know if you could see yourself using any of them, then I'll add them in a separate PR for common (and review). 

Added:
- Simple imputer
- Iterative imputer with random forest (to mimic missforest)
- Standard z-score normalizer that infers numeric columns
- Augmentation with synthetic data using synthcity
- Data-centric processing with cleanlab

@MartinBernstorff, @sarakolding, @bokajgd, @erikperfalk  

The reason for the additional pipeline constructor (using imblearn's `Pipeline`), is that sklearn does not allow one to modify the number of rows in X and Y in a pipeline, but imblearn to the rescue. Their Pipeline is just a subclass of sklearn's which adds functionality for resamplers (such as the one I added for synthetic data) and is a drop-in replacement.

### CODE REVIEW

**Comment by MartinBernstorff** [comment]
Looks interesting! Would love to hear whether they add any performance 👍

File: psycop/projects/scz_bp/model_training/binary_classification_pipeline.py/imblearn_binary_classification_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import pandas as pd
+from imblearn.pipeline import Pipeline
+
+from psycop.common.model_training_v2.config.baseline_registry import BaselineRegistry
+from psycop.common.model_training_v2.trainer.task.base_metric import PredProbaSeries
+
+
+@BaselineRegistry.task_pipelines.register("imblearn_binary_classification_pipeline")
+class ImblearnBinaryClassificationPipeline:
+    def __init__(self, sklearn_pipe: Pipeline):
```
Comment (MartinBernstorff) [inline psycop/projects/scz_bp/model_training/binary_classification_pipeline.py/imblearn_binary_classification_pipeline.py:10]: If `Imblearn.Pipeline` is a subclass of `sklearn.Pipeline`, and does not have invalid overrides, it should be allowed in our `BinaryClassificationPipeline`. Did you get an error, or is there some other reason you needed a separate pipeline class? :-)

File: psycop/projects/scz_bp/model_training/estimator_steps/infer_numeric_standardisation.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        categorical_cols = infer_categorical_features(X)
+        numeric_cols = list(set(X.columns) - set(categorical_cols))  # type: ignore
+        return numeric_cols
+
+    def fit(  # type: ignore
+        self,
+        X: pd.DataFrame,
+        y: None = None,
+        sample_weight: ArrayLike | None = None,
+    ) -> "InferNumericStandardScaler":
```
Comment (MartinBernstorff) [inline psycop/projects/scz_bp/model_training/estimator_steps/infer_numeric_standardisation.py:45]: Could save a bit of time by type-hinting with `Self` :-)

File: .vscode/launch.json [hunk 3]
```diff
@@ -15,5 +15,15 @@
...
+        {
+            "name": "Debug Pytest",
+            "type": "python",
+            "request": "test",
+            "console": "integratedTerminal",
+            // "justMyCode": false,
+            "env": {
+                "PYTHONPATH": "${workspaceFolder}${pathSeparator}${env:PYTHONPATH}"
+            },
+        }
```
Comment (MartinBernstorff) [inline .vscode/launch.json:27]: Probably delete before merge?

File: psycop/projects/scz_bp/model_training/estimator_steps/infer_numeric_standardisation.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import pandas as pd
+from deepchecks.utils.type_inference import infer_categorical_features
+from numpy.typing import ArrayLike
+from sklearn.preprocessing import StandardScaler
+
+from psycop.common.model_training_v2.config.baseline_registry import BaselineRegistry
+from psycop.common.model_training_v2.trainer.task.model_step import ModelStep
+
+
+class InferNumericStandardScaler(StandardScaler):
```
Comment (MartinBernstorff) [inline psycop/projects/scz_bp/model_training/estimator_steps/infer_numeric_standardisation.py:12]: Is z-score normalisation of categorical values represented as int a problem? 
They'd still appear ordinal with constant spacing after normalisation, right?

File: requirements.txt [hunk 5]
```diff
@@ -32,4 +32,7 @@ timeseriesflattener==1.8.0
 wandb==0.15.4
 wasabi==0.9.1
 xgboost==1.7.3
-
+### awaiting approval from Andreas and Søren
+# synthcity==0.2.9 # currently forces a downgrade torch to 1.3.1..
+# imbalanced-learn==0.11.0
+# cleanlab==2.5.0
```
Comment (MartinBernstorff) [inline requirements.txt:38]: If you remove these comments, we can see whether they induce new vulnerabilities :-)

**Review by MartinBernstorff — COMMENTED** [review]
This looks super interesting Lasse! Implementation is on point as well; the new refactor is really paying off here. I.e. it's now possible to test how these augmentations would affect accuracy, even if we don't use them at inference time 👍

File: psycop/projects/scz_bp/model_training/estimator_steps/infer_numeric_standardisation.py (hunk 2 above)
Comment (HLasse) [reply psycop/projects/scz_bp/model_training/estimator_steps/infer_numeric_standardisation.py]: sfsd

**Pull request merged**